packages/commonwealth/server/migrations/20230518155338-remove-offchain-prefix.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (t) => {
+      const renameIndex = async (oldIndexName, newIndexName) => {
+        await queryInterface.sequelize.query(
+          `
+              ALTER INDEX IF EXISTS "${oldIndexName}" RENAME TO "${newIndexName}";
+          `,
+          { transaction: t }
+        );
+      };
+
+      const renameConstraint = async (tableName, oldName, newName) => {
+        await queryInterface.sequelize.query(
```

Comment:
can we do add **IF exists** **constraints**, before trying to rename?

Reply:
if you like we can add this new helper function in out postgres

```
create or replace function create_constraint_if_not_exists (
    t_name text, c_name text, constraint_sql text
) 
returns void AS
$$
begin
    -- Look for our constraint
    if not exists (select constraint_name 
                   from information_schema.constraint_column_usage 
                   where table_name = t_name  and constraint_name = c_name) then
        execute constraint_sql;
    end if;
end;
$$ language 'plpgsql'

```


or just use guard clause to check same.
```
 if not exists (select constraint_name 
                   from information_schema.constraint_column_usage 
                   where table_name = t_name  and constraint_name = c_name)
```

Reply:
As far as I am aware `IF EXISTS` is not something that can be added to constraint renaming operations.

Reply:
yes, thats why add it as enclosing IF

IF exists check BEGIN 
     rename sql
END

Reply:
Your comment did not have all that when I commented lol but yes I can add a guard clause like the existing one for the rest of the queries.